### PR TITLE
Build snippets

### PR DESCRIPTION
**Pull request by wom-bat**
Add build snippets for the serial path

### CODE REVIEW

File: serial/components/serial_components.mk [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+ifeq ($(strip $(SERIAL_NUM_CLIENTS)),)
+$(error Specify the number of clients for the serial virtualisers.  Expect -DSERIAL_NUM_CLIENTS=3 or similar)
+endif
+ifeq ($(strip $(UART_DRIVER)),)
+$(error The serial virtualisers need headers from the UART source. PLease specify UART_DRIVER)
+endif
+
+CFLAGS_serial := -I ${SDDF}/include -I${UART_DRIVER}/include -I${SDDF}/util/include ${SERIAL_NUM_CLIENTS} 
+serial_RX_virtualiser.elf: virt_rx.o
```
Comment (Ivan-Velickovic) [inline serial/components/serial_components.mk:20]:
```suggestion
serial_rx_virt.elf: virt_rx.o
```

**Comment by Ivan-Velickovic** [comment]
Only comment is to use lowercase for the ELF names.

File: serial/components/serial_components.mk (hunk 1 above)
Comment (wom-bat) [reply serial/components/serial_components.mk:20]: Sure.

Comment (wom-bat) [reply serial/components/serial_components.mk:20]: And the corresponding change in tx

File: serial/components/serial_components.mk [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# it should be included into your project Makefile
+#
+# It relies on the variable SERIAL_NUM_CLIENTS to configure the virtualisers
+#
+
+ifeq ($(strip $(SERIAL_NUM_CLIENTS)),)
+$(error Specify the number of clients for the serial virtualisers.  Expect -DSERIAL_NUM_CLIENTS=3 or similar)
+endif
+ifeq ($(strip $(UART_DRIVER)),)
+$(error The serial virtualisers need headers from the UART source. PLease specify UART_DRIVER)
```
Comment (Ivan-Velickovic) [inline serial/components/serial_components.mk:16]:
```suggestion
$(error The serial virtualisers need headers from the UART source. Please specify UART_DRIVER)
```

minor typo

File: serial/components/serial_components.mk [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	$(LD) $(LDFLAGS) $^ $(LIBS) -o $@
+
+virt_rx.o: ${SDDF}/serial/components/virt_rx.c
+	${CC} ${CFLAGS} ${CFLAGS_serial} -o $@ -c ${SERIAL_NUM_CLIENTS} $<
+
+virt_tx.o: ${SDDF}/serial/components/virt_tx.c
+	${CC} ${CFLAGS} ${CFLAGS_serial} -o $@ -c ${SERIAL_NUM_CLIENTS} $<
+
+clean::
+	rm -f mux_[rt]x.[od]
```
Comment (Ivan-Velickovic) [inline serial/components/serial_components.mk:32]: Should this be `virt_*`?

**Comment by Ivan-Velickovic** [comment]
Would you mind also adding a snippet for the i.MX UART?

File: serial/components/serial_components.mk (hunk 3 above)
Comment (wom-bat) [reply serial/components/serial_components.mk:32]: Yes.  I found a heap of faults going back to it; new version about to be pushed.

**Comment by wom-bat** [comment]
OK, I think that fixed all your comments, plus I made the virtualisers rebuild if NUM_SERIAL_CLIENTS or BOARD change.  It builds for me for the Odroid and virtualisers anyway.  Can fix as we use it more for the rest.

**Comment by Ivan-Velickovic** [comment]
> OK, I think that fixed all your comments, plus I made the virtualisers rebuild if NUM_SERIAL_CLIENTS or BOARD change. It builds for me for the Odroid and virtualisers anyway. Can fix as we use it more for the rest.

Cool

**Review by Ivan-Velickovic — APPROVED** [review]

**Pull request merged**